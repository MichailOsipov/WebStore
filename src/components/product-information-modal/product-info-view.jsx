import * as React from 'react';
import {Button} from '@material-ui/core';

export const ProductInfoView = ({
    productInfo,
    onGoShoppingCart,
    onAddProductToCart,
    onGoMainPage
}) => {
    const {name, description} = productInfo || {};
    return (
        <div>
            Информация о товаре:
            {name}<br />
            {description}
            <Button variant="contained" onClick={onAddProductToCart}>Добавить продукт в корзину</Button>
            <Button variant="contained" onClick={onGoShoppingCart}>Перейти в корзину</Button>
            <Button onClick={onGoMainPage}>На главную</Button>
        </div>
    );
};
