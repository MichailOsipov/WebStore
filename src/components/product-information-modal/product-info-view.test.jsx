import * as React from 'react';
import {shallow} from 'enzyme';
import {ProductInfoView} from './product-info-view';

describe('ProductInfoView', () => {
    test('rendering', () => {
        const wrapper = shallow(
            <ProductInfoView
                productInfo={{name: '123', description: '345', imageSrc: '456'}}
                onGoShoppingCart={jest.fn()}
                onAddProductToCart={jest.fn()}
                onGoMainPage={jest.fn()}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
