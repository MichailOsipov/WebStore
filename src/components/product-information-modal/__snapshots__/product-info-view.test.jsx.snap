// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductInfoView rendering 1`] = `
<WithStyles(Grid)
  container={true}
  direction="column"
  spacing={16}
>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(Typography)
      variant="h5"
    >
      Информация о товаре:
    </WithStyles(Typography)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(Card)>
      <WithStyles(CardActionArea)>
        <WithStyles(CardMedia)
          image="456"
          style={
            Object {
              "height": 300,
            }
          }
          title="product-123"
        />
        <WithStyles(CardContent)>
          <WithStyles(Typography)
            component="h2"
            variant="h5"
          >
            123
          </WithStyles(Typography)>
          <WithStyles(Typography)
            component="p"
          >
            345
          </WithStyles(Typography)>
        </WithStyles(CardContent)>
      </WithStyles(CardActionArea)>
      <WithStyles(CardActions)>
        <WithStyles(Button)
          color="primary"
          onClick={[MockFunction]}
        >
          Добавить продукт в корзину
        </WithStyles(Button)>
      </WithStyles(CardActions)>
    </WithStyles(Card)>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(Button)
      onClick={[MockFunction]}
      variant="contained"
    >
      Перейти в корзину
    </WithStyles(Button)>
    <WithStyles(Button)
      onClick={[MockFunction]}
    >
      На главную
    </WithStyles(Button)>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
